### PR TITLE
Feature(Widgets): add a search bar widget

### PR DESCRIPTION
**Pull request by mgermerie**
## Description

Add a search bar widget, allowing user to search locations through a specified geocoding service.

### CODE REVIEW

File: examples/widgets_searchbar.html [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            }
+
+            // Define options for geocoding service that should be used by the searchbar.
+            const geocodingOptions = {
+                url: new URL(
+                    'https://wxs.ign.fr/ayxvok72rcocdyn8xyvy32og/ols/apis/completion?text=&type=StreetAddress,' +
+                    'PositionOfInterest',
+                ),
+                parser: (result) => result.results.map(l => l.fulltext),
+                whenSelected: lookAtCoordinate,
```
Comment (gchoqueux) [inline examples/widgets_searchbar.html]: could you replace `whenSelected` by `onSelected`

File: examples/widgets_searchbar.html [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                }
+            }
+
+            // Define options for geocoding service that should be used by the searchbar.
+            const geocodingOptions = {
+                url: new URL(
+                    'https://wxs.ign.fr/ayxvok72rcocdyn8xyvy32og/ols/apis/completion?text=&type=StreetAddress,' +
+                    'PositionOfInterest',
+                ),
+                parser: (result) => result.results.map(l => l.fulltext),
```
Comment (gchoqueux) [inline examples/widgets_searchbar.html]: could you add a comment to explain the parser?

could you write `response.results.` ?

could you map the coordinates?

File: utils/gui/Searchbar.js [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    fontSize: 16,
+};
+
+
+function addActive(htmlElements, index) {
+    if (!htmlElements) { return index; }
+
+    removeAllActives(htmlElements);
+    if (index >= htmlElements.length) { index = 0; }
+    if (index < 0) { index = (htmlElements.length - 1); }
```
Comment (gchoqueux) [inline utils/gui/Searchbar.js]: `.at(-1)`?

File: utils/gui/Searchbar.js [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+            this.eraseSuggestionList(form);
+
+            currentFocus = -1;
+
+            if (!value) { return false; }
+
+            geocodingOptions.url.searchParams.set('text', value);
+
+            Fetcher.json(geocodingOptions.url).then((geocodingResult) => {
+                const resultArray = geocodingOptions.parser(geocodingResult);
```
Comment (gchoqueux) [inline utils/gui/Searchbar.js]: could you rename `resultArray`?

File: utils/gui/Searchbar.js [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+
+                        // eslint-disable-next-line no-loop-func
+                        autocompleteItem.addEventListener('mouseover', () => {
+                            removeAllActives(form.children);
+                            currentFocus = i;
+                            autocompleteItem.classList.add('active');
+                        });
+
+                        autocompleteItem.addEventListener('click', () => {
+                            geocodingOptions.whenSelected(geocodingResult.results[i]);
```
Comment (gchoqueux) [inline utils/gui/Searchbar.js]: could you use the map instead of `geocodingResult`

File: examples/widgets_searchbar.html (hunk 1 above)
Comment (mgermerie) [reply examples/widgets_searchbar.html]: Renamed.

File: utils/gui/Searchbar.js (hunk 3 above)
Comment (mgermerie) [reply utils/gui/Searchbar.js]: Cannot use `.at` method with `htmlElements` which is of type `HTMLCollection`

File: utils/gui/Searchbar.js (hunk 5 above)
Comment (mgermerie) [reply utils/gui/Searchbar.js]: Done

File: utils/gui/Searchbar.js (hunk 4 above)
Comment (mgermerie) [reply utils/gui/Searchbar.js]: Done

File: examples/widgets_searchbar.html (hunk 2 above)
Comment (mgermerie) [reply examples/widgets_searchbar.html]: I did the changes

**Comment by mgermerie** [comment]
I figured there is an issue with search bar and mouse interaction : when double-clicking the search bar on a `GlobeView`, you trigger a travel animation. On a `PlanarView` it appears the search bar is not clickable. This has to do with event propagation in `GlobeControls` and `PlanarControls`. I shall figure out a way to fix this.

**Review by gchoqueux — APPROVED** [review]

**Pull request merged**